packages/adp-tooling/src/prompts/change-data-source/index.ts

```diff
@@ -0,0 +1,58 @@
+import type { NumberQuestion, ListQuestion, InputQuestion, YUIQuestion } from '@sap-ux/inquirer-common';
+import type { ManifestNamespace } from '@sap-ux/project-access';
+import type { ChangeDataSourceAnswers } from '../../types';
+import { t } from '../../i18n';
+import { filterDataSourcesByType } from '@sap-ux/project-access';
+import { isNotEmptyString } from '../../base/helper';
+
+/**
+ * Gets the prompts for changing the data source.
+ *
+ * @param {Record<string, ManifestNamespace.DataSource>} dataSources - Data sources from the manifest.
+ * @returns {YUIQuestion<ChangeDataSourceAnswers>[]} The questions/prompts.
+ */
+export function getPrompts(
+    dataSources: Record<string, ManifestNamespace.DataSource>
+): YUIQuestion<ChangeDataSourceAnswers>[] {
+    const dataSourceIds = Object.keys(filterDataSourcesByType(dataSources, 'OData'));
+    return [
+        {
+            type: 'list',
+            name: 'id',
+            message: t('prompts.oDataSourceLabel'),
+            choices: dataSourceIds,
+            default: dataSourceIds[0],
+            store: false,
+            guiOptions: {
+                mandatory: true,
+                hint: t('prompts.oDataSourceTooltip')
+            }
+        } as ListQuestion<ChangeDataSourceAnswers>,
+        {
+            type: 'input',
+            name: 'uri',
+            message: t('prompts.oDataSourceURILabel'),
+            guiOptions: {
+                mandatory: true,
+                hint: t('prompts.oDataSourceURITooltip')
+            },
+            validate: isNotEmptyString,
+            when: !!dataSourceIds.length,
+            store: false
+        } as InputQuestion<ChangeDataSourceAnswers>,
+        {
+            type: 'number',
+            name: 'maxAge',
+            message: t('prompts.maxAgeLabel'),
+            when: (answers: ChangeDataSourceAnswers) => answers.uri !== ''
+        } as NumberQuestion<ChangeDataSourceAnswers>,
+        {
+            type: 'input',
+            name: 'annotationUri',
+            message: t('prompts.oDataAnnotationSourceURILabel'),
```

Comment:
Could we please indicate in the label that it is not required to be entered.

Reply:
In Yeoman UI's `guiOptions` all mandatory questions have `mandatory` flag set to `true`. This will render mandatory asterisk in the UI which is sufficient for our use-case.
The annotation URI and max age are optional so they are `mandatory: false` by default.
Maybe the generic prompt converter can add this info to the label in the simple prompt lib case.

Reply:
Good idea, I will contribute it.

Reply:
https://github.com/SAP/open-ux-tools/pull/1925/commits/770da73d49d8f47186debf42d5c0287307f5cdc0